### PR TITLE
OP-2163: added possibility to configure check for delete operation

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OP-2163

- add missing maker-checker logic for deleting group
- make possibility to enable/disable by config - by default maker checker is enabled for Group Deletion

### CODE REVIEW

File: individual/tests/group_service_test.py [hunk 1]
```diff
@@ -118,6 +118,34 @@ def test_update_group_individuals(self):
         # self.assertFalse(individual2.id in individual_ids)
         self.assertTrue(individual3.id in individual_ids)
 
+    def test_delete_group_with_individual(self):
+        individual1 = self.__create_individual()
+        individual2 = self.__create_individual()
+        individual3 = self.__create_individual()
```
Comment (weilu) [inline individual/tests/group_service_test.py]: individual3 is created but never used.

File: individual/gql_mutations.py [hunk 2]
```diff
@@ -337,7 +337,11 @@ def _mutate(cls, user, **data):
         if ids:
             with transaction.atomic():
                 for identifier in ids:
-                    service.delete({'id': identifier})
+                    obj_data = {'id': identifier}
+                    if IndividualConfig.check_group_delete:
+                        service.create_delete_task(obj_data)
+                    else:
+                        service.delete(obj_data)
```
Comment (weilu) [inline individual/gql_mutations.py:344]: @sniedzielski the added test tests the group deletion logic in services but it doesn't cover the branching logic here. Consider adding a gql test to cover this.

File: individual/tests/group_service_test.py (hunk 1 above)
Comment (sniedzielski) [reply individual/tests/group_service_test.py]: thanks for spotting this. I removed this line of code.

File: individual/gql_mutations.py (hunk 2 above)
Comment (sniedzielski) [reply individual/gql_mutations.py:344]: Sure. I added tests fro gql considering row security and general permission (like you did). Now Sonar shows that coverage of new code is 100 %.

**Comment by sniedzielski** [comment]
@weilu Could you re-check PR? I added tests as you proposed in your comments.

**Review by weilu — DISMISSED** [review]

**Comment by weilu** [comment]
@sniedzielski Thanks! I simplified the test to be just sufficient to cover the conditional maker-checker logic. I don't think we need to repeat the coverage on Group delete permission.

**Review by weilu — APPROVED** [review]

**Review by lruzicki — APPROVED** [review]
LGTM

**Comment by sniedzielski** [comment]
Module tests are green, but there is one fail in 'all modules tests':
```
ERROR: test_other_optionset_sync (dhis2_etl.tests.daily_sync_tests.DailySyncTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/core/validation/base.py", line 62, in validator
    instance.full_clean(validate_unique=False)
  File "/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/django/db/models/base.py", line 1502, in full_clean
    raise ValidationError(errors)
django.core.exceptions.ValidationError: {'code': ['Ensure this value has at most 6 characters (it has 55).']}

During handling of the above exception, another exception occurred:
```
@delcroip What should we do with that test? Should we try to fix it before merging this and other PRs?

**Pull request merged**